cost-analyzer/templates/aggregator-cloud-cost-deployment.yaml

```diff
@@ -40,101 +40,15 @@ spec:
             defaultMode: 420
             secretName: {{ .Values.kubecostModel.federatedStorageConfigSecret }}
         {{- end }}
-        {{- if .Values.kubecostProductConfigs.cloudIntegrationSecret }}
         - name: cloud-integration
           secret:
             secretName: {{ .Values.kubecostProductConfigs.cloudIntegrationSecret }}
             items:
               - key: cloud-integration.json
                 path: cloud-integration.json
-        {{- else }}
-        {{- fail "Cloud Cost requires configuration secret" }}
-        {{- end }}
```

Comment:
thinking here being you just rely on the pod waiting in pending for this if they don't supply it?

Reply:
Two reasons: there are some better checks set up in helpers tpl and, more importantly, there are non-secret ways of authing, like workload identity.

Reply:
And yes, per Sean, the cloud-cost container can start up and just do nothing until a config is available without causing a massive failure.